lib/solargraph/pin/base.rb

```diff
@@ -354,7 +354,7 @@ def choose_pin_attr(other, attr)
         end
         # arbitrary way of choosing a pin
         # @sg-ignore Need _1 support
-        [val1, val2].compact.min_by { _1.best_location.to_s }
+        [val1, val2].compact.max_by { File.basename(_1.best_location.to_s) }
```

Comment:
min vs max is arbitrary, but this choice lets me reproduce the original issue on my machine rather than papering it over :)